lld/ELF/SyntheticSections.h

```diff
@@ -138,6 +137,10 @@ class GotSection final : public SyntheticSection {
     bool isSymbolFunc;
   };
   SmallVector<AuthEntryInfo, 0> authEntries;
+
+  // To track GOT entries for symbols that may have been ICFed
```

Comment:
The tuple makes it a bit unclear what the map is, so if we're going to use it so we don't have to stamp out our own DenseMapInfo traits, it would be helpful to elaborate on the key in the comments here. My draft comment text is:

// Map of GOT entries keyed by section, offset, and type. The purpose is to reuse GOT entries when multiple same-type, foldable symbols refer to the image location. In general, this is a GOT-size optimization, but it is also required for some cases involving multi-instruction GOT access patterns and ICF.

Reply:
Thanks. Done